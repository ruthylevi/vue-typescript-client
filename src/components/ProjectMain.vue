<template>
  <div>
    <v-card>
      <v-toolbar flat>
        <v-img :src="avatar" max-height="50" max-width="50"></v-img>
        <v-toolbar-title style=" font-family: Montserrat, font-size: 30px" v-text="title"></v-toolbar-title>
      </v-toolbar>
      <v-img :src="coverImg" contain max-height="350"></v-img>
    </v-card>
    <v-card>
      <v-row>
        <v-col>
          <v-list-item-title class="headline text-left">Description:</v-list-item-title>
          <v-list-item class="text-left" v-text="description"></v-list-item>
          <v-list-item-title class="headline text-left">Majors Required:</v-list-item-title>
          <v-chip-group column>
            <v-chip
              label
              v-for="major in majors"
              :key="major[0]"
              class="noClick"
            >{{major[0] +": "+ major[1]}}</v-chip>
          </v-chip-group>
          <v-list-item-title class="headline text-left">Tags:</v-list-item-title>
          <v-chip-group column>
            <v-chip label v-for="tag in tags" :key="tag" class="noClick">{{tag}}</v-chip>
          </v-chip-group>
        </v-col>
      </v-row>
    </v-card>
  </div>
</template> 

<script>
export default {
  props: {
    avatar: String,
    title: String,
    starred: Boolean,

    coverImg: String,

    description: String,

    majors: Array,
    tags: Array
  }
};
</script>

<style>
.logo {
  width: 50px;
  height: 50px;
}
</style>