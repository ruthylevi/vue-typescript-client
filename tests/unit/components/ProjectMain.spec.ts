import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import { shallowMount } from '@vue/test-utils';
import ProjectMain from '@/components/ProjectMain.vue';

Vue.use(Vuetify);
Vue.use(VueRouter);

const wrapper = shallowMount(ProjectMain);

describe('ProjectMain', () => {
  test('Sanity', () => {
    expect(true).toBe(true);
  });

  test('Visible', () => {
    expect(wrapper.isVisible()).toBe(true);
  });

  test('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
