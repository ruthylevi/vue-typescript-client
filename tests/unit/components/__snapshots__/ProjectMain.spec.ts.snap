// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectMain Renders correctly 1`] = `
<div>
  <v-toolbar-stub tag="header" tile="true" extensionheight="48" src="">
    <!---->
    <h2 class="text-left"></h2>
    <!---->
    <!---->
    <v-spacer-stub></v-spacer-stub>
    <v-item-group-stub activeclass="v-item--active">
      <v-list-item-stub activeclass="" tag="div">
        <v-item-stub></v-item-stub>
      </v-list-item-stub>
    </v-item-group-stub>
    <v-tooltip-stub opendelay="0" closedelay="0" contentclass="" fixed="true" top="true" openonhover="true" maxwidth="175" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" tag="span"> <span> is not accepting applications.</span></v-tooltip-stub>
  </v-toolbar-stub>
  <p></p>
  <v-card-stub loaderheight="4" tag="div">
    <v-img-stub maxheight="350" gradient="to top right, rgba(100,115,201,.33), rgba(25,32,72,.7)" options="[object Object]" position="center center" src="https://picsum.photos/766/350?undefined" transition="fade-transition">
      <v-overlay-stub color="#212121" dark="true" value="true" absolute="true" opacity="0.46" zindex="5">
        <v-container-stub tag="div" class="white--text display-3"></v-container-stub>
      </v-overlay-stub>
    </v-img-stub>
  </v-card-stub>
  <p></p>
  <v-card-stub loaderheight="4" tag="div">
    <v-container-stub tag="div">
      <h2 class="text-left">
        Tagline:
        <!---->
        <!---->
      </h2>
      <v-container-stub tag="div" class="text-left"></v-container-stub>
      <h2 class="text-left">
        Description:
        <!---->
        <!---->
      </h2>
      <v-container-stub tag="div" class="text-left"></v-container-stub>
      <h2 class="text-left">Majors:</h2>
      <v-chip-group-stub activeclass="v-slide-item--active" nexticon="$next" mobilebreakpoint="1264" previcon="$prev" column="true"></v-chip-group-stub> <br>
      <h2 class="text-left">Tags:</h2>
      <v-chip-group-stub activeclass="v-slide-item--active" nexticon="$next" mobilebreakpoint="1264" previcon="$prev" column="true"></v-chip-group-stub>
    </v-container-stub>
  </v-card-stub>
</div>
`;
